transportable-udfs-codegen/src/test/java/com/linkedin/transport/codegen/TestUtils.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.codegen;
+
+import com.linkedin.transport.compile.UDFProperties;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.testng.Assert;
+
+
+class TestUtils {
+
+  private TestUtils() {
+  }
+
+  static UDFProperties getUDFPropertiesFromResource(String resource) {
+    try (InputStreamReader reader = new InputStreamReader(
+        Thread.currentThread().getContextClassLoader().getResourceAsStream(resource))) {
+      return UDFProperties.fromJson(reader);
+    } catch (IOException e) {
+      throw new RuntimeException("Could not read UDF properties from resource: " + resource, e);
+    }
+  }
+
+  /**
+   * Asserts that the contents of both directories (and their subdirectories) are equal
+   */
+  static void assertDirectoriesAreEqual(Path actualDir, Path expectedDir) {
```

Comment:
Sort listings of both directories. Go over sorted lists in parallel. Both names and contents should correspond.
See if you can stick to File all the way in this case.
Check if the chain of calling can be simplified after this approach.